### PR TITLE
Program registration attribute portal

### PR DESCRIPTION
**Pull request by RubenGeo**
[AB#30278](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/30278) 

## Describe your changes

Update the portal to new registration data model

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: interfaces/Portal/src/app/program/registration-activity-detail-accordion/registration-activity-detail-accordion.component.ts [hunk 1]
```diff
@@ -92,7 +92,9 @@ export class RegistrationActivityDetailAccordionComponent implements OnInit {
     );
 
     if (
-      !PaymentUtils.hasVoucherSupport(paymentRow.fsp) &&
+      !PaymentUtils.hasVoucherSupport(
+        paymentRow.programFinancialServiceProviderConfigurationTranslatedLabel,
```
Comment (elwinschmitz) [inline interfaces/Portal/src/app/program/registration-activity-detail-accordion/registration-activity-detail-accordion.component.ts:96]: Shouldn't this be the "non-translatable", enum-name?

So, `paymentRow.programFinancialServiceProviderConfigurationName`? or `financialServiceProviderName`?

File: interfaces/Portal/src/app/program/registration-activity-detail-accordion/registration-activity-detail-accordion.component.ts [hunk 2]
```diff
@@ -114,7 +116,9 @@ export class RegistrationActivityDetailAccordionComponent implements OnInit {
 
     if (
       this.canViewVouchers &&
-      PaymentUtils.hasVoucherSupport(paymentRow.fsp) &&
+      PaymentUtils.hasVoucherSupport(
+        paymentRow.programFinancialServiceProviderConfigurationTranslatedLabel,
```
Comment (elwinschmitz) [inline interfaces/Portal/src/app/program/registration-activity-detail-accordion/registration-activity-detail-accordion.component.ts:120]:
```suggestion
        paymentRow.programFinancialServiceProviderConfigurationName,
```

Same here?

File: services/121-service/src/registration/registrations.service.ts [hunk 3]
```diff
@@ -605,6 +585,16 @@ export class RegistrationsService {
         }
       }
     }
+
+    if (
+      attribute ===
+      AdditionalAttributes.programFinancialServiceProviderConfigurationName
+    ) {
+      registration = await this.updateChosenFspConfiguration({
+        registration: registration,
```
Comment (elwinschmitz) [inline services/121-service/src/registration/registrations.service.ts:594]:
```suggestion
        registration,
```

File: services/121-service/src/registration/registrations.service.ts [hunk 4]
```diff
@@ -817,42 +813,17 @@ export class RegistrationsService {
...
+    //   referenceId,
+    //   relations: ['programFinancialServiceProviderConfiguration'],
+    // });
+    // if (
+    //   registration.programFinancialServiceProviderConfigurationId ===
+    //   newFspConfig.id
+    // ) {
+    //   const errors = `New FSP config id is the same as existing FSP config id for this Person Affected.`;
+    //   throw new HttpException({ errors }, HttpStatus.BAD_REQUEST);
+    // }
```
Comment (elwinschmitz) [inline services/121-service/src/registration/registrations.service.ts:826]: Could this be "just removed"? Or is there another reason this is commented-out?

File: interfaces/Portal/src/app/components/update-fsp/update-fsp.component.ts [hunk 5]
```diff
@@ -73,22 +69,24 @@ export class UpdateFspComponent implements OnInit {
...
   }
 
   public updateChosenFsp() {
     this.programsService
-      .updateChosenFsp(
-        this.referenceId,
+      .updatePaAttribute(
         this.programId,
+        this.referenceId,
+        'programFinancialServiceProviderConfigurationName', // Not sure how this should be done in a strongly typed way in our Portal way of working
```
Comment (elwinschmitz) [inline interfaces/Portal/src/app/components/update-fsp/update-fsp.component.ts:87]: If there is no `enum` with this attribute-name, then this is as good as can be, I think...

Or you'd have to "create string version of the `programFinancialServiceProviderConfigurationName`-property of a specific (return-)type"... (You can ask Copilot that, probably. :D )

File: services/121-service/src/registration/registrations.service.ts (hunk 4 above)
Comment (RubenGeo) [reply services/121-service/src/registration/registrations.service.ts:826]: Yes, make sense. I removed it in one of the other branches. So resolving this for now

**Pull request merged**